nightfall-optimist/src/services/event-queue.mjs

```diff
@@ -43,6 +45,19 @@ export function nextHigherPriorityQueueHasEmptied(priority) {
   });
 }
 
+/**
+This function will push conditionalmakeblock into fastQueue and assures
+no onchain event is executed in parallel with makeblocks . Additonal checks 
+has to be done for makeblocks since following on chain event in order is important
+which is performed and explained in block-assembler service
+*/
+export async function queueBlockAssembler(proposer) {
+  queues[0].push(async () => {
+    await nextHigherPriorityQueueHasEmptied(0); // prevent conditionalmakeblock from running until fastQueue is emptied
```

Comment:
`await` called on queue 0 has no impact , so i'm unclear why it is being called here. We should not need to do a wait on empty check here

Reply:
`priority` is passed as an argument to make function generic to all queues